homeassistant/components/owlet/config_flow.py

```diff
@@ -0,0 +1,188 @@
+"""Config flow for Owlet Smart Sock integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from pyowletapi.api import OwletAPI
+from pyowletapi.exceptions import (
+    OwletCredentialsError,
+    OwletDevicesError,
+    OwletEmailError,
+    OwletPasswordError,
+)
+from pyowletapi.sock import Sock
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_TOKEN,
+    CONF_PASSWORD,
+    CONF_REGION,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_OWLET_EXPIRY, CONF_OWLET_REFRESH, DOMAIN, POLLING_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("region"): vol.In(["europe", "world"]),
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+class OwletConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Owlet Smart Sock."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialise config flow."""
+        self._entry: ConfigEntry
+        self._region: str
+        self._username: str
+        self._password: str
+        self._devices: dict[str, Sock]
+        self.reauth_entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._region = user_input[CONF_REGION]
+            self._username = user_input[CONF_USERNAME]
+            self._password = user_input[CONF_PASSWORD]
+
+            owlet_api = OwletAPI(
+                region=self._region,
+                user=self._username,
+                password=self._password,
+                session=async_get_clientsession(self.hass),
+            )
+
+            await self.async_set_unique_id(self._username.lower())
+            self._abort_if_unique_id_configured()
+
+            try:
+                token = await owlet_api.authenticate()
+                try:
+                    await owlet_api.validate_authentication()
+                    return self.async_create_entry(
```

Comment:
This does not throw any of the exceptions, so instead put this inside of an else in the try except.